vic/drivers/shared_image/src/vic_store.c

```diff
@@ -25,12 +25,14 @@
  *****************************************************************************/
 
 #include <vic_driver_shared_image.h>
+#include <vic_driver_cesm.h>
 
 /******************************************************************************
  * @brief    Save model state.
  *****************************************************************************/
 void
 vic_store(dmy_struct *dmy_current,
+          case_metadata *cmeta,
```

Comment:
we don't want to pass `case_metadata` in to `vic_store`.  Instead, you should set `filenames.statefile = cmeta->caseid`.